augur/filter.py

```diff
@@ -407,24 +418,27 @@ def run(args):
 
         #If didnt find any categories specified, all seqs will be in 'unknown' - but don't sample this!
         if len(seq_names_by_group)==1 and ('unknown' in seq_names_by_group or ('unknown',) in seq_names_by_group):
-            print("WARNING: The specified group-by categories (%s) were not found."%args.group_by,
+            print("WARNING: The specified group-by categories (%s) were not found."%groups,
                   "No sequences-per-group sampling will be done.")
-            if any([x in args.group_by for x in ['year','month']]):
+            if any([x in groups for x in ['year','month']]):
                 print("Note that using 'year' or 'year month' requires a column called 'date'.")
             print("\n")
         else:
             # Check to see if some categories are missing to warn the user
             group_by = set(['date' if cat in ['year','month'] else cat
-                            for cat in args.group_by])
+                            for cat in groups])
             missing_cats = [cat for cat in group_by if cat not in meta_columns]
-            if missing_cats:
+            if missing_cats and ("dummy" not in missing_cats):
                 print("WARNING:")
                 if any([cat != 'date' for cat in missing_cats]):
                     print("\tSome of the specified group-by categories couldn't be found: ",
                           ", ".join([str(cat) for cat in missing_cats if cat != 'date']))
                 if any([cat == 'date' for cat in missing_cats]):
                     print("\tA 'date' column could not be found to group-by year or month.")
                 print("\tFiltering by group may behave differently than expected!\n")
+            elif "dummy" in missing_cats:
+                print("WARNING:\n\tNo group-by categories were specified.", 
+                      "\n\tSequences-per-group sampling will be done using a dummy category.")
```

Comment:
Good catch to check for the dummy group here, but we want the `--subsample-max-sequences` argument to be an acceptable input even if users don't provide `--group-by`, so we don't need to warn users when `--group-by` is not provided. I think this `elif` check will also go away with the modification to the list comprehension I requested above.